### PR TITLE
Backend docs

### LINKED ISSUE #2: Backend Documentation
**Problem**
The backend is well made but not well documented. Makes it difficult to understand what's happening where

**Fix**

- [x] Add documentation to the schema.prisma file, and the server folder. 
- [x] Perform any code cleanup as deemed appropriate

### PR DESCRIPTION
**Pull request by lukejianu**
CLOSES #2. Added documentation to the db folder.

### CODE REVIEW

File: prisma/schema.prisma [hunk 1]
```diff
@@ -1,30 +1,36 @@
 // This is your Prisma schema file,
 // learn more about it in the docs: https://pris.ly/d/prisma-schema
 
+// Boilerplate--we would only change this if we wanted to change how 
+// the Prisma Clienet is generated.
```
Comment (dillydally414) [inline prisma/schema.prisma:5]:
```suggestion
// the Prisma Client is generated.
```

File: prisma/schema.prisma [hunk 2]
```diff
@@ -1,30 +1,36 @@
...
 }
 
+// Specify the DB type and the connection string AKA DB_URL
 datasource db {
   provider             = "mysql"
   url                  = env("DATABASE_URL")
   referentialIntegrity = "prisma"
 }
 
+// Represents a Role.
```
Comment (dillydally414) [inline prisma/schema.prisma:18]: What is a role?

**Review by dillydally414 — COMMENTED** [review]
Some minor issues

File: prisma/schema.prisma [hunk 3]
```diff
@@ -41,22 +47,24 @@ model Account {
...
-  id           String   @id @default(cuid())
+  id           String   @id @default(cuid()) // primary key
   sessionToken String   @unique
-  userId       String
+  userId       String // foreign key
   expires      DateTime
   user         User     @relation(fields: [userId], references: [id], onDelete: Cascade)
 }
 
+// Represents a User--need this for NextAuth and obviously the data it stores.
```
Comment (DavidOB1) [inline prisma/schema.prisma:59]: Sounds a bit sassy.
```suggestion
// Represents a User--need this for NextAuth and the data it stores.
```

File: src/server/db/README.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+## Prisma
+
+---
+
+Prisma is an [ORM](https://stackoverflow.com/questions/1279613/what-is-an-orm-how-does-it-work-and-how-should-i-use-one), which lets us interact with a database without writing SQL queries as a string. This folder contains our Prisma client, which we use to interact with our mySQL database. In this codebase, the client is accessed through the [Context](../router/context.ts). "Context allows us to "bundle" together information to be passed into each API request". For a better understanding of Prisma overall, I recommend following this [guide](https://www.prisma.io/docs/getting-started/quickstart). Since you need a database for this guide, I recommend using PlanetScale, since it's easy to use and it's what Duy used. 
+
+### Contents:
```
Comment (DavidOB1) [inline src/server/db/README.md:7]:
```suggestion
### Contents
```

**Review by DavidOB1 — APPROVED** [review]
Looks pretty epic

File: src/server/db/README.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    ```typescript
+    editUserMutation.mutate({
+        role: userInfo.role,
+        status: userInfo.status,
+        seatAvail: userInfo.seatAvail,
+        ...
+    })
+    ```
+
+2. When edit is called, The input must first pass through the ```z.object```, which enforces type and other constraints on the data. The key part here is if you want to do scheme validation, follow the syntax below, specifying what you want as a ```z.object```, and putting it as the ```input```. tRPC supports Zod, which is why we can pass it to ```input``` as shown below.
```
Comment (ispapa) [inline src/server/db/README.md:63]:
```suggestion
2. When edit is called, the input must first pass through the ```z.object```, which enforces type and other constraints on the data. The key part here is if you want to do schema validation, follow the syntax below, specifying what you want as a ```z.object```, and putting it as the ```input```. tRPC supports Zod, which is why we can pass it to ```input``` as shown below.
```

**Review by ispapa — APPROVED** [review]
Minor typos

File: src/server/db/README.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+})
+```
+---
+### Changing the Schema
+
+The current schema is defined inside the the [schema.prisma](../../../prisma/schema.prisma). This schema is a default NextAuth schema with a few attributes/enums added, so we can store data we care about, like information about drivers and riders.
+
+You can read more about the schema.prisma syntax [here](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model).
+
+If you change something in the schema.prisma file, the change must be reflected in your actual database and prisma has to regenerate the code for the client. You can achieve both of these with a ```yarn prisma db push``` command. This syncs everything back up.
```
Comment (dillydally414) [inline src/server/db/README.md:109]:
```suggestion
If you change something in the schema.prisma file, the change must be reflected in your actual database and prisma has to regenerate the code for the client. You can achieve both of these with a ```yarn db:schema``` command. This syncs everything back up.
```

File: src/server/db/README.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+3. We use the function ```findOne()```, which returns a user in the user table that matches the ID in the where "clause", and only displays the name of the user. 
+    
+    This is roughly equivalent to the SQL query: ```SELECT name FROM user WHERE id = userId;```
+
+---
+
+### Brief Intro to Zod 
+
+[Zod](https://github.com/colinhacks/zod) is used to validate calls to our tRPC routes/endpoints.
```
Comment (dillydally414) [inline src/server/db/README.md:49]:
```suggestion
[Zod](https://zod.dev/) is used to validate input object types in calls to our tRPC routes/endpoints. 
```

**Review by dillydally414 — CHANGES_REQUESTED** [review]
Going to be a contrarian and request changes, not specifically for my review but for all of them. @ispapa @DavidOB1 don't be afraid to request changes or be a little brutal, PRs are your time to tear into each other's work in a "professional" manner - it's how we as a team improve :party-blob:

**Review by lukejianu — COMMENTED** [review]
testing where this goes.

**Review by dillydally414 — DISMISSED** [review]

**Review by devashish9 — APPROVED** [review]
smh

**Review by dillydally414 — APPROVED** [review]
😒 dismissing my stale review.......

**Pull request merged**